tokio/src/park/thread.rs

```diff
@@ -129,6 +129,10 @@ impl Inner {
             return;
         }
 
+        if dur == Duration::from_millis(0) {
+            return;
```

Comment:
This seems like an orthogonal change? Bugfix?

Reply:
I moved some "unit" loom tests to full stack loom tests to expand the coverage. Loom doesn't have a `wait_timeout` implementation (no concept of time), so I needed to add this to make it work.

Reply:
Another option would be to avoid panicking in loom if there is a wait_timeout w/ zero.

Reply:
I'm mostly worried about whether this changes the behavior of existing code. Hard for me to say.

Reply:
It shouldn't. Do you have a suggested change to resolve the comment?

Reply:
I would just have loom implement `wait_timeout` with special-casing for a timeout of 0. Better than possibly changing behavior. But depends on how confident you are that this won't break anything. If you think is the right change, then leave it :)